### PR TITLE
Better compress test

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @proteusvacuum so this should catch most b3 static vs. new_static errors. the code is a bit gross, but it wasn't trivial to get working. very little documentation on it. i did check locally for the negative case. i.e. leave a new_static out and it will fail, and add an if statement to the compress block and it will also fail

fyi: @biyeun 

cc: @emord

### CODE REVIEW

File: corehq/apps/style/tests/test_compress_command.py [hunk 1]
```diff
@@ -1,8 +1,70 @@
...
+
+        in_compress_block = False
+
+        # Filter lines that are not html and strip whitespace
+        filenames = filter(lambda name: name.endswith('.html'), map(lambda name: name.strip(), filenames))
+
+        for filename in filenames:
+            if self._is_b3(filename):
+                with open(filename, 'r+') as f:
+                    for line in f.readlines():
```
Comment (biyeun) [inline corehq/apps/style/tests/test_compress_command.py]: O_O

**Comment by biyeun** [comment]
ouch! but seems reasonable. looks ok to me, will wait on feedback from the others

**Comment by biyeun** [comment]
:beers: for doing this

File: corehq/apps/style/tests/test_compress_command.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/style/tests/test_compress_command.py]: yeah.. it's a lot of parsing, but this test still runs faster than any of our db tests. though that might not be true now with our dependency managed tests

Comment (czue) [inline corehq/apps/style/tests/test_compress_command.py]: :racehorse: :racehorse:

**Comment by czue** [comment]
concept is awesome - thanks for working this out! didn't review the code yet, but happy to defer to others (might not get to this till thursday)

**Comment by proteusvacuum** [comment]
This is great!
Might be useful to also validate whether `hq_shared_tags` was imported if a `new_static` tag was found.

**Comment by proteusvacuum** [comment]
also, not sure how far you want to go, but also would be good to check if there is inline js within the `{% block js %}`, since I think that is a big cause for errors.

**Comment by benrudolph** [comment]
interesting, i'll try and see if there could be something to do about the inline-js thing. i'm gonna punt on the `hq_shared_tags` thing since that'll fail locally if you load the page, i was just a dumbass and didn't load the page before pushing

**Comment by czue** [comment]
I reviewed and looks good. I'm good to ship this but will wait since you mentioned following up. Feel free to self merge if you run out of momentum - think this is a big improvement from nothing.

File: corehq/apps/style/tests/test_compress_command.py [hunk 2]
```diff
@@ -1,8 +1,74 @@
+import re
```
Comment (dimagimon) [inline corehq/apps/style/tests/test_compress_command.py]: F401 're' imported but unused

**Pull request merged**